registry/Scripts/self-signed.ps1

```diff
@@ -1,27 +1,66 @@
-#requires -RunAsAdministrator
+#requires -runasadministrator
+<#
+.Synopsis
+  The script provide functionality to create self signed certificate.
 
+.Description
+  The script provide functionality to create self signed certificate.
+
+.Parameter CertificateCN
+  Fully qualified domain name to create FQDN specific certificate.
+
+.Parameter CertificatePassword
+  Certificate Password to be used to export the cert.
+
+.Parameter CertificateFileExportPath
+  Certificate file export path including certificate filename.
+
+.Example
+   Self-Signed.ps1 -CertificateCN "registry.local.microsoft.com" 
+                   -CertificatePassword <Secret>
+                   -CertificateFileExportPath c:\certificate\registry_cert.pfx
+#>
 Param
 (
-    [Parameter(Mandatory = $true, HelpMessage = "Certificate fully qualified domain name.")]
+    [Parameter(Mandatory = $true, HelpMessage = "Fully qualified domain name to create FQDN specific certificate.")]
     [string] $CertificateCN,
     [Parameter(Mandatory = $true, HelpMessage = "Certificate Password to be used to export the cert.")]
-    [string] $CertificateSecret,
+    [string] $CertificatePassword,
     [Parameter(Mandatory = $true, HelpMessage = "Certificate file export path including certificate filename.")]
     [string] $CertificateFileExportPath
 )
 
 if (-not (Test-Path -Path $CertificateFileExportPath -IsValid))
 {
-    Write-Host "Error: CertificateFileExportPath is not a valid path."
+    throw "Error: CertificateFileExportPath is not a valid path."
 }
 
 if (-not ([IO.Path]::GetExtension($CertificateFileExportPath) -eq '.pfx'))
 {
-    Write-Host "Error: Invalid syntax for CertificateFileExportPath variable. Extension value in path should end with '.pfx'"
+    throw "Error: Invalid syntax for CertificateFileExportPath variable. Extension value in path should end with '.pfx'"
+}
+
+if (Test-Path -Path $CertificateFileExportPath)
+{
+    throw "Error: File already exist ($CertificateFileExportPath). Please remove to provide a different name."
```

Comment:
Error: File ($CertificateFileExportPath) already exist. Please remove it or provide a different name.